client/src/components/donateForm/index.jsx

```diff
@@ -0,0 +1,160 @@
+            console.log('Validate Failed:', info);
+          });
+      }}
+    >
+      <Form
+        form={form}
+        layout="vertical"
+        name="form_in_modal"
+        initialValues={{
+          modifier: 'public',
+        }}
+      >
+
+        <Form.Item
+          name="type"
+          label="Type Of Donation :"
+          rules={[{ required: true, message: 'Please choose donation type!' }]}
+        >
+          <Radio.Group name="type" onChange={({ target: { value } }) => setRadioValue(value)}>
+            <Radio.Button value="clothes">Clothes</Radio.Button>
+            <Radio.Button value="food">Food</Radio.Button>
+            <Radio.Button value="money">Money</Radio.Button>
+          </Radio.Group>
+        </Form.Item>
+        <Form.Item
+          label="Money count/  Piece of clothes/ Number of meals:"
+          name={radioValue}
+          rules={[{ required: true, message: 'Please fill this input!' }]}
+        >
+          <InputNumber min={0} defaultValue={0} name={radioValue} />
+        </Form.Item>
+        <Form.Item
+          name="description"
+          label="Description"
+          rules={[{ required: true, message: 'Please fill this input!' }]}
+        >
+          <Input type="textarea" name="description" />
+        </Form.Item>
+        <Form.Item
+          name="location"
+          label="Location"
+          rules={[{ required: true, message: 'Please fill this input!' }]}
+        >
+          <Input type="textarea" name="location" />
+        </Form.Item>
+        <Form.Item
+          name="deliver_time"
+          label="Deliver Time:"
+          rules={[{ required: true, message: 'Please fill this input!' }]}
+        >
+          <Space name="deliver_time" direction="vertical">
+            <DatePicker
+              selected={selectedDate}
+              name="deliver_time"
+              onChange={(date, dateString) => setSelectedDate(dateString)}
+              dateFormat="dd/MM/yyyy"
+            />
+          </Space>
+        </Form.Item>
+        <Text type="danger">{msgError}</Text>
+      </Form>
+    </Modal>
+  );
+}
+
+function CollectionsPage() {
+  const [visible, setVisible] = useState(false);
+  const [selectedDate, setSelectedDate] = useState(Date.now().toString());
+  const [msgError, setMsgError] = useState();
+  const { id } = useSelector((state) => state.user.userData);
+  const onCreate = async (values) => {
+    try {
+      const { campaignId } = useParams();
+      const donateInfo = {
+        ...values, deliver_time: selectedDate, donorId: id,
+      };
+      const response = await axios.post(`/api/donation/${campaignId}`, donateInfo);
+      setVisible(false);
+      message.success(response.data.message);
+    } catch (error) {
+      setMsgError(error.response.data.message);
+    }
+  };
+
+  return (
+    <>
+      <Button
+        type="primary"
+        onClick={() => {
+          setVisible(true);
+        }}
+      >
+        Donate
```

Comment:
I think the visibility functionality will not be done here. Just do the modal and the logic of donation and Mohammed should connect it to the campaign page